### ISSUE TITLE
[pr-relay] scikit-learn/scikit-learn #34661

### ISSUE THREAD

**Issue by aryansk** [aryansk/pr-relay#33]
{"version":1,"upstream":"scikit-learn/scikit-learn","fork":"aryansk/scikit-learn","base":"main","branch":"fix/34661-svmlight-multilabel-validate","commitMessage":"fix: reject scalar targets in multilabel svmlight dump","prTitle":"FIX prevent svmlight multilabel scalar-target crash","prBody":"#### Reference Issues/PRs\nFixes #34661\n\n#### What does this implement/fix?\n`dump_svmlight_file(..., multilabel=True)` currently lets a one-dimensional scalar target vector reach the low-level multilabel writer, which expects each row target to be iterable and can segfault. Reject that invalid shape at the Python validation boundary with an actionable ValueError.\n\n#### Validation\nAdds focused regression coverage proving a 1D scalar target fails safely instead of entering the native writer.\n\n#### AI usage disclosure\nAI assistance was used for codebase inspection and drafting; the public crash reproducer was used to bound the fix.","patch":"diff --git a/sklearn/datasets/_svmlight_format_io.py b/sklearn/datasets/_svmlight_format_io.py\n--- a/sklearn/datasets/_svmlight_format_io.py\n+++ b/sklearn/datasets/_svmlight_format_io.py\n@@ -580,6 +580,12 @@\n     yval = check_array(y, accept_sparse=\"csr\", ensure_2d=False)\n+    if multilabel and not sp.issparse(yval) and yval.ndim == 1:\n+        if any(np.isscalar(labels) for labels in yval):\n+            raise ValueError(\n+                \"When multilabel=True, y must contain an iterable of labels \"\n+                \"for each sample.\"\n+            )\n     if sp.issparse(yval):\n         if yval.shape[1] != 1 and not multilabel:\n             raise ValueError(\ndiff --git a/sklearn/datasets/tests/test_svmlight_multilabel_validation.py b/sklearn/datasets/tests/test_svmlight_multilabel_validation.py\nnew file mode 100644\n--- /dev/null\n+++ b/sklearn/datasets/tests/test_svmlight_multilabel_validation.py\n@@ -0,0 +1,16 @@\n+import numpy as np\n+import pytest\n+\n+from sklearn.datasets import dump_svmlight_file\n+\n+\n+def test_dump_svmlight_rejects_scalar_targets_for_multilabel(tmp_path):\n+    X = np.eye(2)\n+    y = np.array([0, 1])\n+\n+    with pytest.raises(ValueError, match=\"iterable of labels\"):\n+        dump_svmlight_file(\n+            X,\n+            y,\n+            tmp_path / \"data.svm\",\n+            multilabel=True,\n+        )\n","draft":true,"upstreamIssue":34661}

**Comment by aryansk** [comment]
FAILED
Stage: push
Reason: git failed with exit code 1: ost: github.com
10:54:56.768953 http.c:871              => Send header: User-Agent: git/2.54.0
10:54:56.768955 http.c:871              => Send header: Accept: */*
10:54:56.768958 http.c:871              => Send header: Accept-Encoding: deflate, gzip, br, zstd
10:54:56.768960 http.c:871              => Send header: AUTHORIZATION: basic [redacted]
10:54:56.768962 http.c:871              => Send header: Accept-Language: C, *;q=0.9
10:54:56.768965 http.c:871              => Send header: Pragma: no-cache
10:54:56.768967 http.c:871              => Send header:
10:54:57.104042 http.c:859              <= Recv header, 0000000013 bytes (0x0000000d)
10:54:57.104066 http.c:871              <= Recv header: HTTP/2 200
10:54:57.104072 http.c:859              <= Recv header, 0000000026 bytes (0x0000001…
